### PR TITLE
Add llm endpoint creation and inference sample code to self hosting d…

### PR DESCRIPTION
**Pull request by ruizehung-scale**
Add llm endpoint creation and inference sample code to self hosting doc play with it section.

## Related Issue

https://github.com/scaleapi/llm-engine/issues/141

## Test Plan
```
(launch) ➜  devbox ~ curl -X POST 'http://localhost:5000/v1/llm/model-endpoints' \
    -H 'Content-Type: application/json' \
    -d '{
        "name": "llama-7b",
        "model_name": "llama-7b",
        "source": "hugging_face",
        "inference_framework": "text_generation_inference",
        "inference_framework_image_tag": "0.9.3",
        "num_shards": 4,
        "endpoint_type": "streaming",
        "cpus": 32,
        "gpus": 4,
        "memory": "40Gi",
        "storage": "40Gi",
        "gpu_type": "nvidia-ampere-a10",
        "min_workers": 1,
        "max_workers": 12,
        "per_worker": 1,
        "labels": {},
        "metadata": {}
    }' \
    -u test_user_id:
{"endpoint_creation_task_id":"8d323344-b1b5-497d-a851-6d6284d2f8e4"}%                                                                               
```

```
(launch) ➜  devbox ~ curl -X POST 'http://localhost:5000/v1/llm/completions-sync?model_endpoint_name=llama-7b' \
    -H 'Content-Type: application/json' \
    -d '{
        "prompts": ["Tell me a joke about AI"],
        "max_new_tokens": 30,
        "temperature": 0.1
    }' \
    -u test-user-id:
{"status":"SUCCESS","outputs":[{"text":". Tell me a joke about AI. Tell me a joke about AI. Tell me a joke about AI. Tell me","num_completion_tokens":30}],"traceback":null}%                                                                                                                          
```

### CODE REVIEW

File: docs/guides/self_hosting.md [hunk 1]
```diff
@@ -136,8 +136,68 @@ Forward a port from a `llm-engine` pod:
...
+Next, let's create a LLM endpoint using llama-7b:
+```
+$ curl -X POST 'http://localhost:5000/v1/llm/model-endpoints' \
+    -H 'Content-Type: application/json' \
+    -d '{
+        "name": "llama-7b",
+        "model_name": "llama-7b",
+        "source": "hugging_face",
+        "inference_framework": "text_generation_inference",
+        "inference_framework_image_tag": "0.9.1",
```
Comment (yunfeng-scale) [inline docs/guides/self_hosting.md:158]: we should recommend 0.9.3 which supports llama 2

File: docs/guides/self_hosting.md [hunk 2]
```diff
@@ -136,8 +136,68 @@ Forward a port from a `llm-engine` pod:
...
+        "num_shards": 4,
+        "endpoint_type": "streaming",
+        "cpus": 32,
+        "gpus": 4,
+        "memory": "40Gi",
+        "storage": "40Gi",
```
Comment (yunfeng-scale) [inline docs/guides/self_hosting.md:164]: have you tested whether 40Gi is enough?

File: docs/guides/self_hosting.md [hunk 3]
```diff
@@ -136,8 +136,68 @@ Forward a port from a `llm-engine` pod:
...
+        "gpu_type": "nvidia-ampere-a10",
+        "min_workers": 1,
+        "max_workers": 12,
+        "per_worker": 1,
+        "labels": {"team": "infra", "product": "llm_model_zoo"},
```
Comment (yunfeng-scale) [inline docs/guides/self_hosting.md:169]: remove these labels

File: docs/guides/self_hosting.md [hunk 4]
```diff
@@ -136,8 +136,68 @@ Forward a port from a `llm-engine` pod:
...
+llm-engine-endpoint-id-end-cismpd08agn003rr2kc0-7f86ff64f9qj9xp   2/2     Running   1 (4m41s ago)   7m26s
+```
+Note the endpoint name could be different.
+
+Then, you can send an inference request to the endppoint:
+```
+$ curl -X POST 'http://localhost:5000/v1/llm/completions-sync?model_endpoint_name=llama-7b' \
+    -H 'Content-Type: application/json' \
+    -d '{
+        "prompts": ["hi"],
```
Comment (yunfeng-scale) [inline docs/guides/self_hosting.md:193]: i would have a more exciting prompt

File: docs/guides/self_hosting.md [hunk 5]
```diff
@@ -136,8 +136,68 @@ Forward a port from a `llm-engine` pod:
...
+```
+
+You should get the following response:
+```
+{"model_endpoints":[]}
+```
+
+Next, let's create a LLM endpoint using llama-7b:
+```
+$ curl -X POST 'http://localhost:5000/v1/llm/model-endpoints' \
```
Comment (yixu34) [inline docs/guides/self_hosting.md:151]: Hmm I think we'll want a separate guide that explains the server abstractions, so that people understand what `name` vs `model_name` mean.

Also, I think we want docs per field, but that's more of an API reference, which doesn't need to go here.

File: docs/guides/self_hosting.md [hunk 6]
```diff
@@ -136,8 +136,68 @@ Forward a port from a `llm-engine` pod:
...
+    }' \
+    -u test_user_id:
+```
+
+It should output something like:
+```
+{"endpoint_creation_task_id":"8d323344-b1b5-497d-a851-6d6284d2f8e4"}
+```
+
+Wait a few minutes for the endpoint to be ready. Once it's ready, you can list pods and see `2/2` in the `READY` column:
```
Comment (yixu34) [inline docs/guides/self_hosting.md:180]: subtle nit around causality: Is seeing `READY` an _indicator_ that the endpoint is ready? In other words, I might reword this to "You can tell that it's ready by listing pods and..."

Though alternatively, we may want to `curl` a status endpoint, to keep things at the API level.

File: docs/guides/self_hosting.md [hunk 7]
```diff
@@ -136,8 +136,68 @@ Forward a port from a `llm-engine` pod:
...
+        "prompts": ["hi"],
+        "max_new_tokens": 10,
+        "temperature": 0.1
+    }' \
+    -u test-user-id:
+```
+
+You should get a response similar to:
+```
+{"status":"SUCCESS","outputs":[{"text":"hi hi hi2 hi2 hi2 hi2","num_completion_tokens":10}],"traceback":null}
```
Comment (yixu34) [inline docs/guides/self_hosting.md:202]: Hmm @seanshi-scale @yunfeng-scale I think `status` and `traceback` should be gone?

File: docs/guides/self_hosting.md (hunk 2 above)
Comment (ruizehung-scale) [reply docs/guides/self_hosting.md:164]: Yes! It's tested!

File: docs/guides/self_hosting.md (hunk 6 above)
Comment (ruizehung-scale) [reply docs/guides/self_hosting.md:180]: The `status` in model endpoint record is independent from the status of the pods .... Maybe at some point we might want to add an api that checks the endpoint pod status?

Comment (seanshi-scale) [reply docs/guides/self_hosting.md]: We do have something that returns the number of available/unavailable workers. Not sure if this exposed on the llm endpoint get/list routes though

File: docs/guides/self_hosting.md (hunk 7 above)
Comment (yixu34) [reply docs/guides/self_hosting.md]: Oh oops they're still there in this repo. Will create a separate issue: https://github.com/scaleapi/llm-engine/issues/159

File: docs/guides/self_hosting.md (hunk 6 above)
Comment (yixu34) [reply docs/guides/self_hosting.md]: Oh right nm.

**Review by yixu34 — APPROVED** [review]
